proxy/create_container_interceptor.go

```diff
@@ -71,25 +70,30 @@ func (i *createContainerInterceptor) InterceptRequest(r *http.Request) error {
 	return nil
 }
 
-func (i *createContainerInterceptor) setWeaveWaitEntrypoint(container *docker.Config) error {
-	if len(container.Entrypoint) == 0 {
-		image, err := i.proxy.client.InspectImage(container.Image)
+func (i *createContainerInterceptor) setWeaveWaitEntrypoint(container *createContainerRequestBody) error {
+	config := container.Config
+	if len(config.Entrypoint) == 0 {
+		image, err := i.proxy.client.InspectImage(config.Image)
 		if err == docker.ErrNoSuchImage {
-			return &ErrNoSuchImage{container.Image}
+			return &ErrNoSuchImage{config.Image}
 		} else if err != nil {
 			return err
 		}
 
-		if len(container.Cmd) == 0 {
-			container.Cmd = image.Config.Cmd
+		if len(config.Cmd) == 0 {
+			config.Cmd = image.Config.Cmd
 		}
 
-		if container.Entrypoint == nil {
-			container.Entrypoint = image.Config.Entrypoint
+		if config.Entrypoint == nil {
+			config.Entrypoint = image.Config.Entrypoint
 		}
 	}
 
-	container.Entrypoint = append(weaveWaitEntrypoint, container.Entrypoint...)
+	if len(config.Entrypoint) == 0 || config.Entrypoint[0] != weaveWaitEntrypoint[0] {
+		config.Entrypoint = append(weaveWaitEntrypoint, config.Entrypoint...)
+	}
+	container.HostConfig.VolumesFrom = append(container.HostConfig.VolumesFrom, "weaveproxy")
```

Comment:
I'd rather leave the volume setting where it was, so the function can continue to take just a `*docker.Config`.